### PR TITLE
Get kube agent chart for EKS enrollment without saving temporary files.

### LINKED ISSUE #37959: Discover UI EKS Enrolment fails on Cloud
Currently EKS enrolment requires access to a temporary folder to download Helm chart for the kube agent installation. On Teleport Cloud (tenant `benchmark.cloud.gravitational.io`) I encountered an error related to the read-only filesystem, so chart file could not be saved to a temporary folder.
image

Potential solution:
Manually download chart file into memory and make Helm to work with that data, instead of the file on a filesystem.

### PR DESCRIPTION
**Pull request by AntonAM**
This PR changes the way we prepare kube agent chart before installing it to the enrolled EKS cluster - now we don't save any temporary files in the process. That was causing problems if Teleport was running in the environment where temporary files couldn't be created.

Fixes https://github.com/gravitational/teleport/issues/37959

Changelog: Do not create temporary files during Discover UI EKS cluster enrollment.

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

File: lib/integrations/awsoidc/eks_enroll_clusters.go [hunk 1]
```diff
@@ -503,37 +487,54 @@ type installKubeAgentParams struct {
 	joinToken    string
 	resourceID   string
 	actionConfig *action.Configuration
-	settings     *helmCli.EnvSettings
 	req          EnrollEKSClustersRequest
 	log          logrus.FieldLogger
 }
 
+func getChartUrl(version string) string {
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go]: I think the most straightforward way is still to use `fmt.Sprintf`, but I've moved url.Parse into the `getChartURL`. https://github.com/gravitational/teleport/pull/38606/commits/3de0991a4685746c9712a375bbdb611675934f53

[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go]: Done 👍 https://github.com/gravitational/teleport/pull/38606/commits/1f4d0f917b2c4247c77a47df8c944d2e3f7e27e0

**Review by rosstimothy — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Comment by tigrato** [comment]
Which envs were causing the problems when loading the chart?

**Comment by AntonAM** [comment]
@tigrato Tim noticed it on benchmark.cloud.gravitational.io

**Comment by tigrato** [comment]
> @tigrato Tim noticed it on benchmark.cloud.gravitational.io 

That would be easily fixed by adding a k8s empty dir to the pod manifest

**Pull request merged**